libs/zmath/src/util.zig

```diff
@@ -0,0 +1,169 @@
+// ==============================================================================
+//
+// Collection of useful functions building on top of, and extending, core zmath.
+// https://github.com/michal-z/zig-gamedev/tree/main/libs/zmath
+//
+// ------------------------------------------------------------------------------
+// 1. Matrix functions
+// ------------------------------------------------------------------------------
+//
+// matTranslation(m: Mat) Vec
+// matForward(m: Mat) Vec
+// matUp(m: Mat) Vec
+// matRight(m: Mat) Vec
+//
+//
+// ------------------------------------------------------------------------------
+// 2. Angle functions
+// ------------------------------------------------------------------------------
+//
+// degToRad(angle: f32) f32
+// radToDeg(angle: f32) f32
+// angleMod(angle: f32) f32
+//
+// ==============================================================================
+
+const zm = @import("zmath.zig");
+const std = @import("std");
+const math = std.math;
+const expect = std.testing.expect;
+
+pub fn matTranslation(m: zm.Mat) zm.Vec {
+    return m[3];
+}
+
+pub fn matForward(m: zm.Mat) zm.Vec {
+    return zm.normalize3(zm.mul(zm.f32x4(0.0, 0.0, 1.0, 0.0), m));
+}
+
+pub fn matUp(m: zm.Mat) zm.Vec {
+    return zm.normalize3(zm.mul(zm.f32x4(0.0, 1.0, 0.0, 0.0), m));
```

Comment:
I think it can be optimized to:
`zm.normalize3(zm.f32x4(m[0][1], m[1][1], m[2][1], 0.0));`